datastore/google/cloud/datastore/key.py

```diff
@@ -310,13 +310,25 @@ def to_legacy_urlsafe(self):
         .. note::
 
             The string returned by ``to_legacy_urlsafe`` is equivalent, but
-            not identical, to the string returned by ``ndb``.
+            not identical, to the string returned by ``ndb``. The location
+            prefix may need to be specified to obtain idendentical urlsafe
+            keys.
+
+        :type location_prefix: str
+        :param location_prefix: The location prefix of ones project. Often
+                                this value is 's~', but may be 'dev~', 'e~', or
+                                other location prefixes currently unknown.
 
         :rtype: bytes
         :returns: A bytestring containing the key encoded as URL-safe base64.
         """
+        if location_prefix:
```

Comment:
Can you please make this an explicit [`None`-check][1]:

```python
if location_prefix is None:
    project_id = self.project
else:
    project_id = location_prefix + self.project
```

[1]: http://iruntheinternet.com/lulzdump/images/nunchucks-nuns-chuck-norris-nun-12593488599.jpg?id=